pkg/systemd/services/service.jsx

```diff
@@ -40,147 +41,140 @@
         console.debug.apply(console, arguments); // not-covered: debugging
 }
 
-export class Service extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            unit_id: null,
-            unit_props: null,
-            error: null,
-            reloading: false,
-        };
-
-        this.path = null;
-        this.subscriptions = [];
-    }
-
-    async componentDidMount() {
-        const dbus = this.props.dbusClient;
-        try {
-            const [path] = await dbus.call(s_bus.O_MANAGER, s_bus.I_MANAGER, "LoadUnit", [this.props.unitId]);
-            this.path = path;
-        } catch (ex) {
-            this.setState({ error: ex.toString() });
-            return;
-        }
-
-        this.subscriptions.push(dbus.subscribe(
+export const Service = ({ dbusClient, owner, unitId, unitIsValid, addTimerProperties, pinnedUnits }) => {
+    const [error, setError] = useState(null);
+    const [reloading, setReloading] = useState(false);
+    const [path, setPath] = useState(null);
+    const [unitProps, setUnitProps] = useState(null);
+
+    const updateProperties = () => {
+        const promises = [dbusClient.call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_UNIT])];
+
+        if (unitId.endsWith(".timer"))
+            promises.push(dbusClient.call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_TIMER]));
+        else if (unitId.endsWith(".socket"))
+            promises.push(dbusClient.call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_SOCKET]));
+
+        Promise.all(promises)
+                .then(replies => {
+                    const unit_props = replies[0][0];
+                    const other_props = replies[1]?.[0];
+
+                    // unwrap variants
+                    for (const key in unit_props)
+                        unit_props[key] = unit_props[key].v;
+
+                    if (unitId.endsWith(".timer"))
+                        addTimerProperties(other_props, unit_props);
+                    else if (unitId.endsWith(".socket"))
+                        unit_props.Listen = other_props.Listen.v;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18481-20230310-072502-c32f9a69-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)